src/client/components/UserPage/Drawer/ControlPanel/widgets/DownloadButton.tsx

```diff
@@ -6,12 +6,49 @@ import {
   MenuItem,
   Typography,
 } from '@material-ui/core'
+import FileSaver from 'file-saver'
 
 import TrailingButton from './TrailingButton'
 import downloadIcon from '../assets/download-icon.svg'
-import { useDrawerState, useDrawerDispatch } from '../..'
-import QRCodeModal from '../QRCodeModal'
-import DrawerActions from '../helpers/reducers'
+import { useDrawerState } from '../..'
+import ImageFormat from '../../../../../../shared/util/imageFormat'
+import { postJson } from '../../../../../util/requests'
+
+// Gets file extension from content-type.
+function getFileExtension(format: ImageFormat) {
+  switch (format) {
+    case ImageFormat.SVG:
+      return 'svg'
+    case ImageFormat.PNG:
+      return 'png'
+    case ImageFormat.JPEG:
+      return 'jpeg'
+    default:
+      throw Error('Invalid format passed to getFileExtension')
+  }
+}
+
+// Downloads QR code from server.
+async function downloadServerQrCode(
+  shortLink: string,
+  format: ImageFormat,
+): Promise<void> {
+  if (typeof window === 'undefined') {
+    throw Error('This code should only run in a browser environment')
+  }
```

Comment:
I don't think this check is necessary given that the code clearly lives in the client-side of the codebase

Reply:
That makes sense, I will remove it